src/backend/distributed/operations/worker_shard_copy.c

```diff
@@ -259,6 +311,26 @@ ShardCopyDestReceiverStartup(DestReceiver *dest, int operation, TupleDesc
 	copyDest->columnOutputFunctions = ColumnOutputFunctions(inputTupleDescriptor,
 															copyOutState->binary);
 	copyDest->copyOutState = copyOutState;
+	CreateReplicationOriginIfNotExists(copyDest);
+}
+
+
+static void
+CreateReplicationOriginIfNotExists(ShardCopyDestReceiver *dest)
+{
+	int connectionFlags = OUTSIDE_TRANSACTION;
+	char *currentUser = CurrentUserName();
+	WorkerNode *workerNode = FindNodeWithNodeId(dest->destinationNodeId,
+												false /* missingOk */);
+	MultiConnection *connection = GetNodeUserDatabaseConnection(connectionFlags,
+																workerNode->workerName,
+																workerNode->workerPort,
+																currentUser,
+																NULL /* database (current) */);
+	ClaimConnectionExclusively(connection);
```

Comment:
maybe omit this call. It's a specific hack we use for opening multiple connections to a node in the executor, which seems to be getting copy-pasted into other places.

Reply:
Ok then I will check if the slot name not containing "citus_shard_move_slot_"  or "citus_shard_split_slot_" to determine if it is a CDC slot.

Reply:
Omitted this call.